data_safe_haven/external/api/azure_sdk.py

```diff
@@ -1234,6 +1239,23 @@ def set_blob_container_acl(
             msg = f"Failed to set ACL '{desired_acl}' on container '{container_name}'."
             raise DataSafeHavenAzureError(msg) from exc
 
+    def storage_exists(
+        self,
+        storage_account_name: str,
+    ) -> bool:
+        """Find out whether a storage account exists in Azure storage
+
+        Returns:
+            bool: Whether or not the storage account exists
+        """
+
+        storage_client = StorageManagementClient(
+            self.credential(), self.subscription_id
+        )
+        return not storage_client.storage_accounts.check_name_availability(
+            {"name": storage_account_name}
+        ).name_available
```

Comment:
Is this better than doing

```python
return storage_account_name in storage_client.storage_account.list
```

[docs](https://learn.microsoft.com/en-us/python/api/azure-mgmt-storage/azure.mgmt.storage.v2023_05_01.operations.storageaccountsoperations?view=azure-python#azure-mgmt-storage-v2023-05-01-operations-storageaccountsoperations-list)